node/utils/subscriptions.ts

```diff
@@ -0,0 +1,62 @@
+const FREQUENCY_PATTERN = /(\d{1,3})?\s*(\w*?)(?:$|s|ly)/i
+
+const SUBSCRIPTION_PREFIX = `vtex.subscription`
+const SUBSCRIPTION_KEY_PREFIX = `${SUBSCRIPTION_PREFIX}.key`
+
+const SUBSCRIPTION_KEY_FREQUENCY = `${SUBSCRIPTION_KEY_PREFIX}.frequency`
+
+export function parseFrequency(frequency: string) {
+  const match = frequency.match(FREQUENCY_PATTERN)
+
+  if (!match) {
+    return
+  }
+
+  const [, count = '1', type] = match
+
+  if (!type) return
+
+  return {
+    interval: +count,
+    periodicity: type.toUpperCase(),
+  } as SubscriptionDataEntry['plan']['frequency']
+}
+
+export function generateSubscriptionDataEntry(
+  subscriptionsFromAssemblyOptions: Array<{
+    itemIndex: number
+    options: AssemblyOptionInput[]
+  }>
+): SubscriptionDataEntry[] {
+  const planType = 'RECURRING_PAYMENT'
+  const defaultExecutionCount = 0
+
+  const subscriptionDataEntries = subscriptionsFromAssemblyOptions.map(item => {
+    const { itemIndex } = item
+
+    const planFrequency = parseFrequency(
+      item.options?.[0].inputValues[SUBSCRIPTION_KEY_FREQUENCY]
```

Comment:
Magic number 0. Is it always there?

Reply:
Yes! It should always be there because this `item.options` object surely contains a subscription. Buut, I see that it could be in a different index for some reason... I've changed the code a bit to make it safer.